src/modsim2/similarity/metrics/pad.py

```diff
@@ -0,0 +1,322 @@
+        self.__embed_train_data = embedding_fn(self.__train_data)
+        self.__embed_test_data = embedding_fn(self.__test_data)
+
+    def _pre_process_data(self, data_A: np.ndarray, data_B: np.ndarray):
+        """
+        Takes A & B datasets as arguments and calls the methods to process
+        the data ahead of building the classifiers. First the data are split
+        into train and test datasets, then labelled and concatenated and then
+        transformed using the embedding function.
+
+        Args:
+            data_A: the records in the A dataset (excludes target values)
+            data_B: the records in the B dataset (excludes target values)
+        """
+        # Split A and B into train and test datasets
+        # This is done before the data are concatenated in order
+        # to ensure that the same proportion of A & B are
+        # included in the test dataset
+        train_A, test_A, train_B, test_B = self.__train_test_split(data_A, data_B)
+        # transform data - will depend on classifier type as to the transform
+
+        self.__concat_data(train_A, test_A, train_B, test_B)
+
+        self.__transform_data()
+
+    def __build_models(
+        self, kernel_name: str, c_values: list, gamma_values: list, degree_values: list
+    ):
+        """
+        This method will build the classifiers, variation in some hyperparameters
+        is allowed as the optimal values are not necessarily known for the datasets.
+        The kernel is not varied.
+        The trained model are appended to the object's list of models.
+
+        Args:
+            kernel_name:
+            c_values: list of C values (regularisation param) to be applied
+            gamma_values: list of gamma values (kernel coefficient) to be applied
+            degree_values: list of degree values for the poly kernel to be applied
+        """
+        for c in c_values:
+            for gamma in gamma_values:
+                for degree in degree_values:
+                    svc = SVC(kernel=kernel_name, C=c, gamma=gamma, degree=degree)
+                    svc.fit(X=self.__embed_train_data, y=self.__train_labels)
+                    self.__models.append(svc)
+
+    def __evaluate_models(self):
+        """
+        This method will evaluate the classifiers that have been built
+        The evaluation metric is the mean absolute error.
+        """
+        for svc in self.__models:
+            preds = svc.predict(self.__embed_test_data)
+            errors = np.abs(preds - self.__test_labels)
+            error = np.sum(errors) / len(preds)
+            self.__errors.append(error)
+
+    def calculate_distance(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        labels_A: np.ndarray,
+        labels_B: np.ndarray,
+        c_values: list,
+        kernel_name: str,
+        embedding_name: str,
+        test_proportion: float,
+        gamma_values: list = ["scale"],
```

Comment:
If set to 'scale' then SVC uses 1 / (n_features * X.var()) as the value of gamma. It is the default value given in sklearn so I used it as the default here too (a default in a list is required so that the code to built the model will loop over it). Gamma can also be provided as a float or 'auto' which will set the value to 1/n_features.